src/saveload/saveload.cpp

```diff
@@ -2680,14 +2680,15 @@ static const SaveLoadFormat _saveload_formats[] = {
  * uncompressed, or another type
  * @param full_name Name of the savegame format. If empty it picks the first available one
  * @param compression_level Output for telling what compression level we want.
- * @return Pointer to SaveLoadFormat struct giving all characteristics of this type of savegame
+ * @return Reference to SaveLoadFormat struct giving all characteristics of this type of savegame
  */
-static const SaveLoadFormat *GetSavegameFormat(const std::string &full_name, uint8_t *compression_level)
+static const SaveLoadFormat &GetSavegameFormat(const std::string &full_name, uint8_t *compression_level)
```

Comment:
`compression_level` could be a reference, as the code seems to assume it's never `nullptr`